### PR TITLE
feat: omega handles iff and implications

### PR DESCRIPTION
**Pull request by semorrison**
While looking at theorems that "ought" to be proved by `omega`, I found a number of `iff` goals where one would need `constructor <;> (intros; omega)` to close the goal. This PR slightly powers up `omega` so it can do this by itself.

### CODE REVIEW

File: Std/Logic.lean [hunk 1]
```diff
@@ -612,7 +612,8 @@ theorem Decidable.not_iff [Decidable b] : ¬(a ↔ b) ↔ (¬a ↔ b) := by
 theorem Decidable.iff_not_comm [Decidable a] [Decidable b] : (a ↔ ¬b) ↔ (b ↔ ¬a) := by
   rw [@iff_def a, @iff_def b]; exact and_congr imp_not_comm not_imp_comm
 
-theorem Decidable.iff_iff_and_or_not_and_not [Decidable b] : (a ↔ b) ↔ (a ∧ b) ∨ (¬a ∧ ¬b) :=
+theorem Decidable.iff_iff_and_or_not_and_not {a b : Prop} [Decidable b] :
+    (a ↔ b) ↔ (a ∧ b) ∨ (¬a ∧ ¬b) :=
```
Comment (semorrison) [inline Std/Logic.lean:616]: This previously had implicit arguments `{b a}`.

**Review by joehendrix — APPROVED** [review]

File: Std/Tactic/Omega/Frontend.lean [hunk 2]
```diff
@@ -245,30 +243,45 @@ def addIntInequality (p : MetaProblem) (h y : Expr) : OmegaM MetaProblem := do
     problem := ← (p.problem.addInequality lc.const lc.coeffs
       (some do mkAppM ``le_of_le_of_eq #[h, (← prf)])) |>.solveEqualities }
 
+
+
```
Comment (digama0) [inline Std/Tactic/Omega/Frontend.lean:247]:
```suggestion
```

File: test/false_or_by_contra.lean [hunk 3]
```diff
@@ -18,7 +18,7 @@ example (h : False) : ¬ P := by
   guard_target = False
   simp_all
 
-example (h : False) : P := by
+example {P  : Prop} (h : False) : P := by
```
Comment (digama0) [inline test/false_or_by_contra.lean:21]: ?

Comment (semorrison) [reply test/false_or_by_contra.lean]: I've made the test clearer. `false_or_by_contra` behaves differently when the goal is a Type vs a Prop. Previously when `P` was an autoimplicit, it was being specialised to `Prop` *by the tactic* in the proof.

**Pull request merged**